.github/workflows/auto-merge.yml

```diff
@@ -0,0 +1,21 @@
+# SPDX-FileCopyrightText: 2022 The Aalto Grades Developers
+#
+# SPDX-License-Identifier: MIT
+
+name: Enable auto-merge
+run-name: Auto-merge ${{ github.actor }} pull request
+on:
+  pull_request:
+    types: [opened, ready_for_review]
```

Comment:
It should work so that when a PR is opened or marked as ready for review, github-actions immediately enables auto-merge on that PR, so that when all the requirements are fulfilled the PR is automatically merged.